framework/src/main/java/org/checkerframework/common/returnsreceiver/ReturnsReceiverAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,98 @@
+package org.checkerframework.common.returnsreceiver;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ElementKind;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.returnsreceiver.qual.BottomThis;
+import org.checkerframework.common.returnsreceiver.qual.This;
+import org.checkerframework.common.returnsreceiver.qual.UnknownThis;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/** The type factory for the Returns Receiver Checker. */
+public class ReturnsReceiverAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+    /** The {@code @}{@link This} annotation. */
+    final AnnotationMirror THIS_ANNOTATION;
```

Comment:
Why does this need to be package private? Could it be protected or even private?

Reply:
I see one usage outside this package, which could just use `areSameByClass`...
at least add a comment why you used package private.